crates/astria-composer/src/executor/mod.rs

```diff
@@ -681,11 +681,12 @@ impl Future for SubmitFut {
                         .nonce(*this.nonce)
                         .chain_id(&*this.chain_id)
                         .build();
-                    let tx = UnsignedTransaction {
-                        actions: this.bundle.clone().into_actions(),
-                        params,
-                    }
-                    .into_signed(this.signing_key);
+                    let tx = UnsignedTransaction::builder()
+                        .actions(this.bundle.clone().into_actions())
+                        .params(params)
+                        .build()
+                        .expect("failed to build transaction from actions")
```

Comment:
I don't know if it is acceptable for a malformed transaction to crash this service / if this function call is behind a protected endpoint.

Reply:
The code in the `bundle_factory` is restricted to just work for `SequenceActions` so this shouldn't ever fail, might be better to move the construction of the `UnsignedTransaction` to the bundle code or to have the `bundle_factory` code to keep everything as explicit `SequenceActions`

Reply:
Went the route of putting it into a function